test/testsyncengine.cpp

```diff
@@ -906,6 +906,108 @@ private slots:
+        fakeFolder.localModifier().insert("A/newFile");
+
+        QVERIFY(!fakeFolder.applyLocalModificationsAndSync());
+        lock.close();
+
+        // we still have the clashed item
+        QVERIFY(fakeFolder.currentLocalState().find(QStringLiteral("A")));
+        QVERIFY(fakeFolder.currentLocalState().find(lockedFileName));
+
+        // the unaffected download happened
+        QVERIFY(fakeFolder.currentLocalState().find(QStringLiteral("B/bb")));
+
+        // the remove did not happen as we had a case clash
+        QVERIFY(fakeFolder.currentLocalState().find(QStringLiteral("A")));
+
+        // S was deleted independent of the clas of A/aa and A/aA
+        QVERIFY(!fakeFolder.currentLocalState().find(QStringLiteral("S")));
+    }
+
+    // why is S deleted in this test
+    void testProceedWithIndependentDelets2()
+    {
+        QFETCH_GLOBAL(Vfs::Mode, vfsMode);
+        QFETCH_GLOBAL(bool, filesAreDehydrated);
+
+        FakeFolder fakeFolder(FileInfo::A12_B12_C12_S12(), vfsMode, filesAreDehydrated);
+        fakeFolder.remoteModifier().mkdir(QStringLiteral("A/JustAFolder"));
+        fakeFolder.remoteModifier().insert(QStringLiteral("A/JustAFolder/WithFiles"));
+        QVERIFY(fakeFolder.applyLocalModificationsAndSync());
+        QCOMPARE(fakeFolder.currentLocalState(), fakeFolder.currentRemoteState());
+
+
+        QVERIFY(fakeFolder.currentLocalState().find(QStringLiteral("A")));
+        QVERIFY(fakeFolder.currentLocalState().find(QStringLiteral("B")));
+        QVERIFY(fakeFolder.currentLocalState().find(QStringLiteral("S")));
+
+        const auto lockedFileName = QStringLiteral("A/JustAFolder/locked");
+        fakeFolder.localModifier().insert(lockedFileName);
+
+        QVERIFY(fakeFolder.applyLocalModificationsAndSync());
+
+        // lock the file to trigger a sync error
+        auto lock = FileSystem::lockFile(fakeFolder.localPath() + lockedFileName, FileSystem::LockMode::Exclusive);
+        QVERIFY(lock);
+
+        fakeFolder.remoteModifier().insert(QStringLiteral("B/bb"));
+        fakeFolder.remoteModifier().remove(QStringLiteral("S"));
+
+        fakeFolder.remoteModifier().remove(QStringLiteral("A"));
+        // ensure we have a job in addition to the remove
+        fakeFolder.localModifier().insert("A/JustAFolder/newFile");
+
+        QVERIFY(!fakeFolder.applyLocalModificationsAndSync());
+        lock.close();
+
+        // we still have the clashed item
+        QVERIFY(fakeFolder.currentLocalState().find(QStringLiteral("A")));
+        QVERIFY(fakeFolder.currentLocalState().find(lockedFileName));
+
+        // the unaffected download happened
+        QVERIFY(fakeFolder.currentLocalState().find(QStringLiteral("B/bb")));
+
+        // the remove did not happen as we had a case clash
+        QVERIFY(fakeFolder.currentLocalState().find(QStringLiteral("A")));
+
+        // S was deleted independent of the clas of A/aa and A/aA
+        QVERIFY(!fakeFolder.currentLocalState().find(QStringLiteral("S")));
```

Comment:
Why is S gone in this case

Reply:
Relevant code 
https://github.com/owncloud/client/blob/bfa988779de6fc08edf905f0d3780dd3db893af2/src/libsync/owncloudpropagator.cpp#L1159-L1163

Reply:
Implemented in:
https://github.com/owncloud/client/commit/1e652e12b5027afa11e7a0443ec4b10669d4d79e

>Note that this means that if there are errors in subJobs the
>dirDeletionJobs won't get executed.